sdk/go/auto/remote_workspace_test.go

```diff
@@ -111,8 +117,25 @@ func testRemoteStackGitSourceErrors(t *testing.T, fn func(ctx context.Context, s
 	}
 }
 
+// fetchCommitHash runs `git ls-remote URL branch` to determine the latest commit for the given repo
+// URL and branch.
+func fetchCommitHash(url, branch string) (string, error) {
+	cmd := exec.Command("git", "ls-remote", url, branch)
+	output, err := cmd.Output()
+	if err != nil {
+		return "", err
```

Comment:
I would add some context to this; otherwise if this fails, we'll just get "exit status 1".
```suggestion
		return "", fmt.Errorf("git ls-remote: %w", err)
```